### PR TITLE
Bug 1977920: Ensure scc compatibility with BoundServiceAccountTokenVolume

### PR DESCRIPTION
**Pull request by stlaz**
Previous to the BoundServiceAccountTokenVolume feature being enabled, the automatic mounting of legacy token secrets required that an scc permit secret volume sources either implicitly (by allowing all volume sources) or explicitly (by specifying 'secret' in the set of allowed volumes).

To ensure compatibility with this permission scheme for the projected token volumes enabled by BoundServiceAccountTokenVolume, this commit ensures that the projected volumes of service account tokens will be permitted under the same criteria (i.e. secret volume sources are allowed by an scc).

Credits also to @marun for the largest part of the changes here.

### CODE REVIEW

File: pkg/securitycontextconstraints/sccmatching/provider_test.go [hunk 1]
```diff
@@ -1004,6 +1007,66 @@ func TestValidateAllowedVolumes(t *testing.T) {
...
+			projectedVolumeSource: serviceAccountTokenVolumeSource(),
+			wantAllow:             false,
+		},
+		{
+			desc:           "deny if the projected volume has volume source other than the ones in projected volume injected by service account token admission plugin",
+			allowedFSTypes: []securityv1.FSType{securityv1.FSTypeSecret},
+			projectedVolumeSource: &api.ProjectedVolumeSource{
+				Sources: []api.VolumeProjection{
+					{
+						ConfigMap: &api.ConfigMapProjection{
```
Comment (sttts) [inline pkg/securitycontextconstraints/sccmatching/provider_test.go:1034]: would like to see any variation of the serviceAccountTokenVolumeSource, i.e. combinations where most is like in a service account, but one detail is changed in each test case.

Comment (stlaz) [reply pkg/securitycontextconstraints/sccmatching/provider_test.go:1033]: added a bunch of unit tests to the set

File: pkg/securitycontextconstraints/util/util.go [hunk 2]
```diff
@@ -176,3 +176,41 @@ func EqualStringSlices(a, b []string) bool {
...
+		if s.ServiceAccountToken == nil && s.ConfigMap == nil && s.DownwardAPI == nil {
+			return false
+		}
+		if t := s.ServiceAccountToken; t != nil && (t.Path != "token" || t.Audience != "") {
+			return false
+		}
+
+		if s.ConfigMap != nil {
+			switch cmRef := s.ConfigMap.LocalObjectReference.Name; cmRef {
+			case "kube-root-ca.crt":
```
Comment (sttts) [inline pkg/securitycontextconstraints/util/util.go:197]: `case "kube-root-ca.crt", "openshift-service-ca.crt":`. Now you fall through to line 207.

File: pkg/securitycontextconstraints/util/util.go [hunk 3]
```diff
@@ -176,3 +176,41 @@ func EqualStringSlices(a, b []string) bool {
...
+			case "openshift-service-ca.crt":
+				if len(s.ConfigMap.Items) != 1 || s.ConfigMap.Items[0].Key != "service-ca.crt" || s.ConfigMap.Items[0].Path != "service-ca.crt" {
```
Comment (sttts) [inline pkg/securitycontextconstraints/util/util.go:199]: `if len(s.ConfigMap.Items) != 1 || (s.ConfigMap.Items[0].Key != "service-ca.crt" && s.ConfigMap.Items[0].Path != "service-ca.crt") {`

File: pkg/securitycontextconstraints/util/util.go (hunk 2 above)
Comment (stlaz) [reply pkg/securitycontextconstraints/util/util.go:221]: added a check specific to `kube-root-ca.crt`

**Pull request merged**